views/js/e2e/runner/navigation/linearTestNavigation.spec.js

```diff
@@ -0,0 +1,346 @@
+            cy.logout();
+        });
+
+        /**
+         * Setup to have a proper delivery:
+         * - Start server
+         * - Add necessary routes
+         * - Guest login
+         * - Start test
+         */
+        beforeEach(() => {
+            cy.setupServer();
+            cy.addRoutes();
+            cy.addTestRoutes();
+            cy.guestLogin();
+            cy.startTest('Delivery of e2e navigation linear test');
+        });
+
+        /**
+         * Destroy everything we created during setup, leaving the environment clean for next time.
+         */
+        after(() => {
+            cy.setupServer();
+            cy.addRoutes();
+            cy.login('admin');
+            cy.deleteItem('e2e navigation linear test');
+            cy.deleteTest('e2e navigation linear test');
+            cy.deleteDelivery('Delivery of e2e navigation linear test');
+        });
+
+        describe('First Item', () => {
+
+            it('Has proper navigation buttons', function () {
+                cy.get('.navi-box-list').within(() => {
+                    // visible navigation buttons
+                    cy.get(setupSelectors.testNavigation.nextItem).should('exist').and('be.visible');
+                    cy.get(setupSelectors.testNavigation.skipItem).should('exist').and('be.visible');
+
+                    // not visible navigation buttons
+                    cy.get(setupSelectors.testNavigation.previousItem).should('exist').and('not.be.visible');
+
+                    //not existing buttons
+                    cy.get(setupSelectors.testNavigation.endTest).should('not.exist');
+                    cy.get(setupSelectors.testNavigation.skipAndEndTest).should('not.exist');
+                });
+            });
+
+            it('Move to next item', cy.nextItem);
+
+            it('Skip to next item', cy.skipItem);
+
+        });
+
+        describe('Second Item', () => {
+
+            it('Has proper navigation buttons', function () {
+                // go to 2nd item
+                cy.nextItem();
+
+                cy.get('.navi-box-list').within(() => {
+                    // visible navigation buttons
+                    cy.get(setupSelectors.testNavigation.nextItem).should('exist').and('be.visible');
+                    cy.get(setupSelectors.testNavigation.skipItem).should('exist').and('be.visible');
+
+                    // not visible navigation buttons
+                    cy.get(setupSelectors.testNavigation.previousItem).should('exist').and('not.be.visible');
+
+                    //not existing buttons
+                    cy.get(setupSelectors.testNavigation.endTest).should('not.exist');
+                    cy.get(setupSelectors.testNavigation.skipAndEndTest).should('not.exist');
+                });
+            });
+
+            it('Move to next item', () => {
+                // go to last item
+                cy.nextItem();
+                cy.nextItem();
```

Comment:
Not straightforward, the title is saying move to next item but we move to the last item.

Reply:
I will change the comments to be more comprehensive.